### PR TITLE
Update design doc for WebApi v7 w.r.t. changed default settings

### PR DESCRIPTION
**Pull request by biaol-odata**
### Issues

*This pull request fixes issue #1490. This is the md format file equivalent to the original docx format document we went thru in design meeting on May 2nd, 2018*

### Description

*Update design doc for WebApi v7 for updated default settings.*
- Support for KeyAsSegment by default,
- Support for Case-insensitive match for Url by default with best case-sensitive match.

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+
+
+Overview
+========
+
+OData Layer
+-----------
+
+OData libraries 7.x (current version 7.4.4) contains updates to improve
+usability & compatibility of the library by virtual of exposing options
```
Comment (mikepizzo) [inline _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md]: >virtual [](start = 44, length = 7)

virtue

File: _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+-   As usual, namespace is the primary mechanism for resolving name
+    token conflicts in multiple schema component of the model, therefore
+    namespace is required up to OData v4. To improve flexibility, with
+    notion of [Default
+    Namespaces](http://docs.oasis-open.org/odata/odata/v4.01/cs01/part1-protocol/odata-v4.01-cs01-part1-protocol.html#sec_DefaultNamespaces)
+    introduced in OData v4.01, namespace qualifier is optional for
+    function or action identifier in request Uri. When corresponding
+    option in ODL Uri parser enabled:
+
+    -   If the function or action identifier containing a namespace
```
Comment (mikepizzo) [inline _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md]: >containing [](start = 45, length = 10)

contains

File: _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+>
+> ����string�functionCallTokenName�=�this.state.Configuration.EnableCaseInsensitiveUriFunctionIdentifier�?�functionCallToken.Name.ToLowerInvariant()�:�functionCallToken.Name;
+
+To implement with the correct behavior for enabled case-insensitive:
+
+-   GetUriFunctionSignatures needs to additionally return signatures associated with function names in a dictionary instance.
+
+-   When resolving best match function based on arguments for the invoked function, MatchSignatureToUriFunction will find the best match. Exception is still thrown in case of ambiguity or no matches found.
+
+Work Items
```
Comment (mikepizzo) [inline _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md]: Work items should only reference work in public github repo.  We should not reference internal urls.

Comment (biaol-odata) [reply _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md]: Updated to reference GH links. Thanks.

---
In reply to: [197311712](https://github.com/OData/WebApi/pull/1492#discussion_r197311712) [](ancestors = 197311712)

File: _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md [hunk 4]
```diff
@@ -0,0 +1,285 @@
+---
+layout: post
+title: "12.1 WebApi 7.0 Default Setting Updates"
+description: "Desgin document for updated default settings for WebApi 7.0"
```
Comment (AlanWong-MS) [inline _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md]: Desgin -> Design

File: _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+# WebApi Default Enable Unqualified Operations and Case-insensitive Uri #
+
+
+Overview
+========
+
+OData Layer
+-----------
+
+OData libraries 7.x (current version 7.4.4) contains updates to improve
```
Comment (AlanWong-MS) [inline _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md]: I think the doc will be outdated as soon as we release 7.5.0. Should we just leave it at 7.x? Or say 7.5.0+?

File: _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+
+### Case-Insensitive name resolution:
+
+-   Case-insensitive property name should become resolvable:
+
+    With model: public class InvalidQueryCustomer { public int <span style="color:red">Id</span> { get; set; } }
+
+    > GET /service/InvalidQueryCustomers?\$filter=<span style="color:blue">id</span> eq 5 : HTTP 200
+
+    >GET /service/InvalidQueryCustomers(5)?\$filter=<span style="color:blue">id</span> eq 5 : HTTP 400 "Query
```
Comment (AlanWong-MS) [inline _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md]: nit: missing space between > and GET

File: _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+
+> public�static�IContainerBuilder�AddDefaultODataServices(this�IContainerBuilder�builder)
+>
+> {
+>
+> //.........
+>
+> builder.AddService(ServiceLifetime.Singleton,�
+>
+> sp�=\>�ODataUriResolver.GetUriResolver(null));
```
Comment (AlanWong-MS) [inline _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md]: Looks like =\> should be =>?

File: _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+        >
+        > ����//�Set�Uri�resolver�to�by�default�enabling�unqualified�functions/actions�and�case�insensitive�match.
+        >
+        > ����builder.AddService(
+        >
+        > ��������ServiceLifetime.Singleton,
+        >
+        > ��������typeof(ODataUriResolver),
+        >
+        > ��������sp�=\>�new�UnqualifiedODataUriResolver�{EnableCaseInsensitive�=�true});
```
Comment (AlanWong-MS) [inline _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md]: Same comment regarding =\>

File: _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+
+    public�ODataUriParser(IEdmModel�model,�Uri�relativeUri,�IServiceProvider�container)
+
+Enable Case-Insensitive for Custom Uri function
+-----------------------------------------------
+
+One issue is encountered when trying to bind function call token with
+case-insensitive enabled. The reason is that at the very beginning of
+the function BindAsUriFunction() the name token, when case-insensitive
+is enabled, is coerced to lower case (as shown blow), which is valid for
```
Comment (AlanWong-MS) [inline _posts/2018-06-18-12-02-WebApiV7_newDefaultSettings.md]: typo: shown blow -> shown below

**Comment by biaol-odata** [comment]
Thanks for the CR.
Closed as merged.

**Pull request merged**